### PR TITLE
imageupload tests: run serially

### PR DESCRIPTION
**Pull request by akalenyu**
We've [recently](https://github.com/kubevirt/kubevirt/pull/12925#issuecomment-2481198061) switched the imageupload suite to run in parallel, which has introduced flakiness:  
https://search.ci.kubevirt.io/?search=unexpected+return+value+502&maxAge=48h&context=1&type=bug%2Bissue%2Bjunit&name=&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job
this was also seen in a previous attempt to make them parallel:
 https://github.com/kubevirt/kubevirt/pull/11716#issuecomment-2373256509

let's put them back to running serial for now as the gain in runtime is not drastic.

### What this PR does
Before this PR:

After this PR:

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Review by ShellyKa13 — APPROVED** [review]
/lgtm

**Review by EdDev — APPROVED** [review]

File: tests/virtctl/imageupload.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ const (
 	pvcSize = "100Mi"
 )
 
-var _ = Describe("[sig-storage][virtctl]ImageUpload", decorators.SigStorage, func() {
+var _ = Describe("[sig-storage][virtctl]ImageUpload", decorators.SigStorage, Serial, func() {
```
Comment (iholder101) [inline tests/virtctl/imageupload.go:37]:
```suggestion
var _ = Describe("[sig-storage][Serial][virtctl]ImageUpload", decorators.SigStorage, Serial, func() {
```

This have also existed here originally: [`36be728` (#12925)](https://github.com/kubevirt/kubevirt/pull/12925/commits/36be728df5618e16bfd78a5ea244d3e46623de05#diff-167e9ebe2dd43d34a0f106feeae260a41a00d7cecdc9dc8cc4cd5aea654c08a1L49)

**Review by iholder101 — COMMENTED** [review]
Thanks a lot for the quick response @akalenyu!

Did you figure out what have caused the flakiness? If so, can you please add the reasoning to the PR description?

**Comment by akalenyu** [comment]
> Thanks a lot for the quick response @akalenyu!
> 
> Did you figure out what have caused the flakiness? If so, can you please add the reasoning to the PR description?

So what I know is that running these in parallel increases flakiness, I am not sure why. Definitely something for us to look at..

File: tests/virtctl/imageupload.go (hunk 1 above)
Comment (akalenyu) [reply tests/virtctl/imageupload.go:37]: Actually that's on purpose. We're pretty [unhappy](https://github.com/kubevirt/kubevirt/pull/13194#issuecomment-2459496047) duplicating the serial decorator. In fact, so unhappy that I've contributed the change to ginkgo - https://github.com/onsi/ginkgo/pull/1484

Comment (iholder101) [reply tests/virtctl/imageupload.go:37]: Awesome :)
Waiting for your PR to consume the new release and remove all `[Serial]`s :wink:

**Comment by iholder101** [comment]
> > Thanks a lot for the quick response @akalenyu!
> > Did you figure out what have caused the flakiness? If so, can you please add the reasoning to the PR description?
> 
> So what I know is that running these in parallel increases flakiness, I am not sure why. Definitely something for us to look at..

Fair enough. Thank you!
/approve

**Comment by 0xFelix** [comment]
I see the retry mechanism I added to virtctl is not enough. Shouldn't have done that in the first place.

To me it still looks like the underlying issue is: https://github.com/kubevirt/containerized-data-importer/issues/3450

**Comment by akalenyu** [comment]
> I see the retry mechanism I added to virtctl is not enough. Shouldn't have done that in the first place.
> 
> To me it still looks like the underlying issue is: [kubevirt/containerized-data-importer#3450](https://github.com/kubevirt/containerized-data-importer/issues/3450)

I wasn't aware of that. The theory looks really promising. WDYT @mhenriks

**Pull request merged**